packages/components/src/components/form/Autocomplete/Autocomplete.tsx

```diff
@@ -56,15 +56,25 @@ export const Autocomplete: FC<AutocompleteProps> & AutocompleteStaticMembers = (
   );
 };
 
-const AutocompleteInput: FC<AutocompleteInputProps> = ({ showClearButton, ...rest }) => {
+const AutocompleteInput: FC<AutocompleteInputProps> = ({
+  showClearButton,
+  showClearButtonWhen = 'item-selected',
```

Comment:
Would you consider other namings like `clearButtonMode` or `clearButtonCriteria` ?

Reply:
Maybe we should consider removing the `showClearButton` and have an enum for the `clearButtonCriteria` or `clearButtonMode` with values:

- `item-selected`: Show the clear button when an item is selected.
- `has-value`: Show the clear button when there is input value.
- `never`: Never show the clear button.

Reply:
Yeah. I've thought about removing it, but removing a prop would be a breaking change.

Should we still do it? WDYT?

Reply:
I'd prefer to have just one property to deal with. Let's do a major